common/include/pcl/experimental/executor/omp_executor.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#ifdef _OPENMP
+  #include <omp.h>
+#endif
+
+#include <pcl/experimental/executor/property.h>
+#include <pcl/experimental/executor/type_trait.h>
+
+namespace executor {
+
+template <typename Blocking, typename ProtoAllocator>
+struct omp_executor;
+
+#ifdef _OPENMP
+template <>
+struct is_executor_available<omp_executor> : std::true_type {};
+#endif
+
+template <typename Blocking = blocking_t::always_t,
+          typename ProtoAllocator = std::allocator<void>>
+struct omp_executor {
+  using shape_type = std::size_t;
```

Comment:
doesn't omp force this to be a signed integer type? should we make it a a pcl::sindex_t?

Reply:
The shape which represents the number of threads should always be positive in OMP IIRC.
I have updated it to`uindex_t`.

Reply:
We don't want users passing negative shapes else we will have to handle that explcitly